### PR TITLE
frontend redeem contribution link

### LINKED ISSUE #1924: 🚀 [Feature] Allow Users to Redeem Contribution Links
## 🚀 Feature
Like Transaction Links, as user must be able to redeem creation links in the wallet frontend. Use as many components as possible from the transaction link mechanism

### PR DESCRIPTION
**Pull request by ogerly**
### Issues
- fixes #1924

### CODE REVIEW

File: frontend/src/pages/TransactionLink.spec.js [hunk 1]
```diff
@@ -210,6 +211,11 @@ describe('TransactionLink', () => {
     describe('token in store and own link', () => {
       beforeEach(() => {
         mocks.$store.state.token = 'token'
+        wrapper.setData({
```
Comment (Tirokk) [inline frontend/src/pages/TransactionLink.spec.js:214]: This testfile makes an error:

```bash
 RUNS  src/pages/TransactionLink.spec.js
/Users/wolfgang/Documents/Programming/Gradido/gradido/frontend/test/testSetup.js:96
  throw new Error(w);
        ^

Error: Error in nextTick: "Error: Error in render: "TypeError: Cannot read properties of undefined (reading 'firstName')""
    at Object.<anonymous>._vue.default.config.warnHandler (/Users/wolfgang/Documents/Programming/Gradido/gradido/frontend/test/testSetup.js:96:9)
    at warn (/Users/wolfgang/Documents/Programming/Gradido/gradido/frontend/node_modules/vue/dist/vue.runtime.common.dev.js:619:26)
    at logError (/Users/wolfgang/Documents/Programming/Gradido/gradido/frontend/node_modules/vue/dist/vue.runtime.common.dev.js:1880:5)
    at globalHandleError (/Users/wolfgang/Documents/Programming/Gradido/gradido/frontend/node_modules/vue/dist/vue.runtime.common.dev.js:1875:3)
    at handleError (/Users/wolfgang/Documents/Programming/Gradido/gradido/frontend/node_modules/vue/dist/vue.runtime.common.dev.js:1835:5)
    at Array.<anonymous> (/Users/wolfgang/Documents/Programming/Gradido/gradido/frontend/node_modules/vue/dist/vue.runtime.common.dev.js:1978:9)
    at flushCallbacks (/Users/wolfgang/Documents/Programming/Gradido/gradido/frontend/node_modules/vue/dist/vue.runtime.common.dev.js:1902:14)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

File: frontend/src/pages/ResetPassword.vue [hunk 2]
```diff
@@ -88,7 +88,11 @@ export default {
             ? this.$t('message.checkEmail')
             : this.$t('message.reset')
           this.messageButtonText = this.$t('login')
-          this.messageButtonLinktTo = '/login'
+          if (this.$route.params.code) {
+            this.messageButtonLinktTo = `/login/${this.$route.params.code}`
+          } else {
+            this.messageButtonLinktTo = '/login'
+          }
```
Comment (Tirokk) [inline frontend/src/pages/ResetPassword.vue:95]: Why ever, with me that does not work on Safari for a new registration with a transaction link.

**Review by Tirokk — DISMISSED** [review]
Hey, great work.
But stil a little thing:

If I use a transaction link and do register as new user get an email with code and set my password everything is okay.

But aufter setting the password and clicking on the login button and come to the login page the code in the URL vanishes and I can not reedem directly the transaction …

File: frontend/src/pages/ResetPassword.vue (hunk 2 above)
Comment (Mogge) [reply frontend/src/pages/ResetPassword.vue:95]: We have to fix this in the future.

Comment (Mogge) [reply frontend/src/pages/ResetPassword.vue:95]: I created an issue

**Review by Mogge — CHANGES_REQUESTED** [review]

**Review by Mogge — APPROVED** [review]

**Pull request merged**